internal/am/job_tracker.go

```diff
@@ -105,16 +152,18 @@ func (jt *JobTracker) savePreservationTasks(ctx context.Context, jobs []amclient
 	return count, nil
 }
 
-// filterSavedJobs filters out jobs that have an ID in saved then returns the
-// filtered job list.
-func filterSavedJobs(jobs []amclient.Job, saved map[string]struct{}) []amclient.Job {
-	var unsaved []amclient.Job
+// filterSavedJobs filters out jobs that have an ID in saved and jobs without
+// LinkID in the jobs we want to keep, then returns the filtered job list.
+func (jt *JobTracker) filterSavedJobs(jobs []amclient.Job) []amclient.Job {
```

Comment:
I think this method should be renamed to `filterJobs()` or something else that doesn't imply that it's only filtering out saved jobs.